src/store/prompts/prompts.selectors.ts

```diff
@@ -112,6 +112,27 @@ export const selectFilteredFolders = createSelector(
   },
 );
 
+export const selectSectionFolders = createSelector(
+  [
+    (state) => state,
+    selectFolders,
+    (_state, filters: EntityFilters) => filters,
+  ],
+  (state, folders, filters) => {
+    const folderIds = folders // direct parent folders
+      .filter((folder) => filters.sectionFilter(folder))
+      .map((folder) => folder.id);
+
+    const filteredFolderIds = new Set(
+      folderIds.flatMap((fid) =>
+        getChildAndCurrentFoldersIdsById(fid, folders),
+      ),
+    );
+
+    return folders.filter((folder) => filteredFolderIds.has(folder.id));
+  },
+);
+
```

Comment:
It might be worth reusing the selector logic, since it's exactly the same as in `conversations.selectors.ts`